### PR TITLE
Show start and end cycle

### PR DESCRIPTION
**Pull request by friedger**
This PR
* adds start and end cycle and pool reward address 
  * to the pooled stacking info
  * to the self-service-extend page
* refactors self-service layout
* shows end cycle in red if extension is required

Screenshot from 2023-06-19 15-34-37

### CODE REVIEW

File: src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  address,
+  stackerInfoDetails,
+  onClose,
+  isContractCallExtensionPageOpen,
+  setShowExtendForOtherUser,
+}: Props) {
+  const { network } = useStacksNetwork();
+  const client = new StackingClient(address, network);
+  const getDelegationStatusQuery = useGetConfirmedDelegationStatus(client);
+  const getDelegationStatusQuery2 = useDelegationStatusForUserQuery({ client, address, network });
```
Comment (kyranjamie) [inline src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx:36]: can you think of a better name?

**Review by kyranjamie — APPROVED** [review]
`(StackerInfo & { stacked: true })['details']` appears 7 times, suggest making a reusable type


thx Friedger 👍🏼

File: src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx (hunk 1 above)
Comment (friedger) [reply src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx:36]: something like useTemporaryWorkaroundUntilStatusApiHasBeenFixedQuery :-)
I prefer to fix stacks.js and then remove the second query again.

**Review by kyranjamie — APPROVED** [review]

File: src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx (hunk 1 above)
Comment (friedger) [reply src/pages/stacking/self-service-extend/components/extend-for-current-user.tsx:37]: Actually, it has been fixed. I am improve the queries now..

**Review by kyranjamie — APPROVED** [review]

**Comment by markmhx** [comment]
cc @314159265359879

**Pull request merged**